lib/embulk/input/twitter_ads_analytics.rb

```diff
@@ -226,33 +229,67 @@ def get_access_token
       end
 
       def request_entities(access_token)
-        url = "https://ads-api.twitter.com/9/accounts/#{@account_id}/#{entity_plural(@entity).downcase}"
-        url = "https://ads-api.twitter.com/9/accounts/#{@account_id}" if @entity == "ACCOUNT"
-        response = access_token.request(:get, url)
-        if response.code != "200"
-          Embulk.logger.error "#{response.body}"
-          raise
+        retries = 0
+        begin
+          url = "https://ads-api.twitter.com/9/accounts/#{@account_id}/#{entity_plural(@entity).downcase}"
+          url = "https://ads-api.twitter.com/9/accounts/#{@account_id}" if @entity == "ACCOUNT"
+          response = access_token.request(:get, url)
+          if response.code != "200"
+            Embulk.logger.error "#{response.body}"
+            raise
+          end
+          return [JSON.parse(response.body)["data"]] if @entity == "ACCOUNT"
+          JSON.parse(response.body)["data"]
+        rescue StandardError => e
+          if retries < NUMBER_OF_RETRIES
+            retries += 1
+            sleep_sec = exp_backoff_sec(response: response, retries: retries)
+            if sleep_sec > MAX_SLEEP_SEC_NUMBER
+              raise e
+            end
+            sleep sleep_sec
+            Embulk.logger.warn("retry #{retries}, #{e.message}")
+            retry
+          else
+            Embulk.logger.error("exceeds the upper limit retry, #{e.message}")
+            raise e
+          end
         end
-        return [JSON.parse(response.body)["data"]] if @entity == "ACCOUNT"
-        JSON.parse(response.body)["data"]
       end
 
       def request_stats(access_token, entity_ids, chunked_time)
-        params = {
-          entity: @entity,
-          entity_ids: entity_ids.join(","),
-          metric_groups: @metric_groups.join(","),
-          start_time: chunked_time[:start_time],
-          end_time: chunked_time[:end_time],
-          placement: @placement,
-          granularity: @granularity,
-        }
-        response = access_token.request(:get, "https://ads-api.twitter.com/9/stats/accounts/#{@account_id}?#{URI.encode_www_form(params)}")
-        if response.code != "200"
-          Embulk.logger.error "#{response.body}"
-          raise
+        retries = 0
+        begin
+          params = {
+            entity: @entity,
+            entity_ids: entity_ids.join(","),
+            metric_groups: @metric_groups.join(","),
+            start_time: chunked_time[:start_time],
+            end_time: chunked_time[:end_time],
+            placement: @placement,
+            granularity: @granularity,
+          }
+          response = access_token.request(:get, "https://ads-api.twitter.com/9/stats/accounts/#{@account_id}?#{URI.encode_www_form(params)}")
+          if response.code != "200"
+            Embulk.logger.error "#{response.body}"
+            raise
```

Comment:
Pleas define a concrete Standard Error class and raise/catch it.

Reply:
fixed by https://github.com/trocco-io/embulk-input-twitter_ads_analytics/pull/6/commits/25bd52add26f2b6e35ba737b4a7ab832a3ec9b21